src/lib/AnnotationControlsFSM.ts

```diff
@@ -41,38 +43,52 @@ export const stateModeMap = {
 export default class AnnotationControlsFSM {
     private currentState: AnnotationState;
 
+    private subscriptions: Subscription[] = [];
+
     constructor(initialState = AnnotationState.NONE) {
         this.currentState = initialState;
```

Comment:
Should this also call `setState`? If so, the separate calls to `updateDiscoverabilityResinTag` wouldn't be needed.

Reply:
It could, but I feel like the `subscribe` is more like a change listener while setting an initial state in the constructor is just that -- more of an initialization